src/Fixer/AttributeNotation/OrderedAttributesFixer.php

```diff
@@ -0,0 +1,349 @@
+                    array_map(static fn (string $attribute): string => ltrim($attribute, '\\'), $value),
+                )))
+                ->getOption(),
+        ]);
+    }
+
+    protected function applyFix(\SplFileInfo $file, Tokens $tokens): void
+    {
+        $index = 0;
+
+        while (null !== $index = $tokens->getNextTokenOfKind($index, [[T_ATTRIBUTE]])) {
+            /** @var list<array{name: string, start: int, end: int}> $elements */
+            $elements = $this->getAttributeDeclarationElements($tokens, $index);
+            $endIndex = end($elements)['end'];
+
+            foreach ($elements as &$element) {
+                $element['name'] = $this->sortAttributes($tokens, $element['start']);
+            }
+
+            try {
+                if (1 === \count($elements)) {
+                    continue;
+                }
+
+                $sortedElements = $this->sortElements($elements);
+
+                if ($elements === $sortedElements) {
+                    continue;
+                }
+
+                $this->sortTokens($tokens, $index, $endIndex, $sortedElements);
+            } finally {
+                $index = $endIndex;
+                $this->clearNamespaceAnalysis();
+            }
+        }
+    }
+
+    /**
+     * Find all consecutive elements that start with #[ and end with ] beginning at $startIndex.
+     *
+     * @return list<array{start: int, end: int}>
+     */
+    private function getAttributeDeclarationElements(Tokens $tokens, int $startIndex): array
+    {
+        $elements = [];
+
+        $index = $startIndex;
+        do {
+            $declarationStartIndex = $index;
+            $declarationEndIndex = $tokens->getNextTokenOfKind($declarationStartIndex, [[CT::T_ATTRIBUTE_CLOSE]]);
+
+            $index = $tokens->getNextMeaningfulToken($declarationEndIndex);
+
+            $elements[] = [
+                'start' => $declarationStartIndex,
+                'end' => $index - 1,
+            ];
+        } while ($tokens[$index]->isGivenKind(T_ATTRIBUTE));
+
+        return $elements;
+    }
+
+    private function sortAttributes(Tokens $tokens, int $startIndex): string
+    {
+        $elements = $this->getAttributeElements($tokens, $startIndex);
+
+        if (1 === \count($elements)) {
+            return $elements[0]['name'];
+        }
+
+        $sortedElements = $this->sortElements($elements);
+
+        if ($elements === $sortedElements) {
+            return $elements[0]['name'];
+        }
+
+        $this->sortTokens($tokens, $startIndex + 1, end($elements)['end'], $sortedElements, new Token(','));
+
+        return $sortedElements[0]['name'];
+    }
+
+    /**
+     * Find all attributes inside #[] beginning at $startIndex.
+     *
+     * @return list<array{name: string, start: int, end: int}>
+     */
+    private function getAttributeElements(Tokens $tokens, int $startIndex): array
```

Comment:
I believe this should go to `PhpCsFixer\Tokenizer\Analyzer\AttributeAnalyzer` to be reusable in other attribute-related fixers. Maybe it's good idea to represent results with some kind of object (collection with items inside), probably classes should go to `src/Tokenizer/Analyzer/Analysis`.

Reply:
Done